### ISSUE TITLE
PrintStackTrace rule does not detect all violations

### ISSUE THREAD

**Issue by davidburstrom** [detekt/detekt#9345]
The problem likely occurs because neither the rule PrintStackTrace::visitCallExpression nor PrintStackTrace::visitCatchSection call through to the super class, so the AST visit is cut short.

## Expected Behavior

Whenever `printStackTrace` is called, it should be reported.

## Observed Behavior

Depending on the syntax tree, `printStackTrace` may or may not be reported.

## Steps to Reproduce

```kotlin
fun whatever() {
    run {
        try {
            check(false
        } catch (e: IllegalStateException) {
            e.printStackTrace() // <-- not flagged, because of the `run` invocation.
        }
    }
    try {
       check(false)
    } catch (e: IllegalStateException) {
        e.printStackTrace() // <-- flagged
    }
}
```

## Your Environment

* Version of Detekt used: 1.23.8

**Comment by cortinico** [comment]
Hey @davidburstrom 

This should be an easy fix if someone wants to pick this us. A good starting point would be to add a failing test using the snippet you just provided
